exec/pipeline/financials/pull_data.py

```diff
@@ -0,0 +1,74 @@
+from prefect import flow, task
+from pydantic import BaseModel
+import requests
+import instructor
+from pydantic import BaseModel, validator
+from openai import OpenAI
+import tempfile
+from decimal import Decimal
+from openai import OpenAI
+
+import PyPDF2
+
+
+class CurrentAssets(BaseModel):
+    cash_and_cash_equivalents: Decimal
+    marketable_securities: Decimal
+    accounts_receivable_net: Decimal
+    vendor_non_trade_receivables: Decimal
+    other_current_assets: Decimal
+    total_current_assets: Decimal
+
+    # @validator("total_current_assets")
+    # def total_current_assets_is_sum_of_all_assets(cls, v):
+    #     return v == (
+    #         cls.cash_and_cash_equivalents
+    #         + cls.marketable_securities
+    #         + cls.accounts_receivable_net
+    #         + cls.vendor_non_trade_receivables
+    #         + cls.other_current_assets
+    #     )
+
+
+class FinancialStatement(BaseModel):
+    current_assets: CurrentAssets
+
+
+@task
+def read_remote_pdf(url, filetype: str = "pdf") -> str:
+    response = requests.get(url)
+
+    with tempfile.NamedTemporaryFile(mode="wb", delete=False) as f:
+        f.write(response.content)
+        reader = PyPDF2.PdfReader(f.name)
+
+    text = ""
+    for page in reader.pages:
+        text += page.extract_text()
+
+    print(text)
+    return text
+
+
+@task
+def parse_financial_statements(data: str) -> FinancialStatement:
+    client = instructor.from_openai(OpenAI())
```

Comment:
I assume the authentication was done via local environment variables?

Reply:
yea, it expects a `OPENAI_API_KEY` in the env, let's make this explicit